tutorials/encode-videos-using-serverless-jobs/index.mdx

```diff
@@ -0,0 +1,134 @@
+---
+meta:
+    title: Encode videos using Serverless Jobs and FFMPEG
+    description: In this tutorial you will learn how to deploy a Serverless Job that encodes videos using FFMPEG
+content:
+    h1: Encode videos using Serverless Jobs and FFMPEG
+    paragraph: In this tutorial you will learn how to deploy a Serverless Job that encodes videos using FFMPEG
+tags: serverless jobs ffmpeg video encoding
+categories:
+  - container-registry
+  - jobs
+  - object-storage
+dates:
+    validation: 2024-05-15
+    posted: 2024-05-15
+---
+
+This tutorial demonstrates the process of encoding videos retrieved from Object Storage using Serverless Jobs: media encoding is a resource-intensive tasks over prolonged durations, making it suitable for Serverless Jobs. The job takes a video file as its input, encodes it using a Docker image based on [FFMPEG](https://ffmpeg.org/), then uploads the encoded video back to the S3 bucket.
+
+<Macro id="requirements" />
+
+- A Scaleway account logged into the [console](https://console.scaleway.com)
+- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
+- An [Object Storage bucket](/storage/object/how-to/create-a-bucket/) 
+- A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
+
+
+## Creating the job image
+
+The initial step involves defining a Docker image for interacting with the S3 Object Storage using [MinIO](https://min.io/) and performing a video encoding task using [FFMPEG](https://ffmpeg.org/).
+
+1. Create a bash script `encode.sh` with the following content:
+    ```bash
+    #!/bin/sh
+    set -e
+
+    echo "Configuring S3 access for MinIO"
+    mc config host add scw "https://$JOB_S3_ENDPOINT/" "$JOB_S3_ACCESS_KEY" "$JOB_S3_SECRET_KEY"
+
+    echo "Downloading the file from S3"
+    mc cp "scw/$JOB_INPUT_PATH/$JOB_INPUT_FILENAME" "/tmp/$JOB_INPUT_FILENAME"
```

Comment:
@antoinechampion 2 issues with this command:

- `mc cp` apparently does not allow S3 to file system copy, whereas `mc get` does
- the source path does not contain the name of the bucket

Could we update the command to the following ?

```
mc get "scw/$BUCKET_NAME/$JOB_INPUT_PATH/$JOB_INPUT_FILENAME" "/tmp/$JOB_INPUT_FILENAME"
```
And add a $BUCKET_NAME variable to the Job's environment variables ?

(I tested with `mc get` and it works fine)

Reply:
I double-checked, `mc cp` does work. Looks like you are running those commands locally, however the tutorial executes them within a container having a fixed and precise execution environment. So it might be due to local environment setup.

Reply:
@antoinechampion The mc cp error occurred in the container and appeared in the logs, I'll try again

Reply:
@antoinechampion tested again and it works, issue must have been linked to the syntax you changed !